tlm_teamd/values_store.h

```diff
@@ -0,0 +1,73 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include <jansson.h>
+
+#include <dbconnector.h>
+
+using StringPair = std::pair<std::string, std::string>;
+using Records = std::unordered_map<std::string, std::string>;
+using HashOfRecords = std::unordered_map<std::string, Records>;
+
+class ValuesStore
+{
+public:
+    ValuesStore(const swss::DBConnector * db) : m_db(db) {};
+    void update(const std::vector<StringPair> & dumps);
+
+private:
+    enum class json_type
+    {
+        string,
+        boolean,
+        integer,
+    };
+
+    json_t * load_json(const std::string & data);
+    std::vector<std::string> get_ports(json_t * root);
+    std::pair<std::vector<std::string>, std::string> convert_path(const std::string & path);
+    json_t * traverse(json_t * root, const std::vector<std::string> & path_array, const std::string & path);
+    std::string unpack_string(json_t * root, const std::string & key, const std::string & path);
+    std::string unpack_boolean(json_t * root, const std::string & key, const std::string & path);
+    std::string unpack_integer(json_t * root, const std::string & key, const std::string & path);
+    std::string get_value(json_t * root, const std::string & path, ValuesStore::json_type type);
+    HashOfRecords from_json(const std::vector<StringPair> & dumps);
+    std::vector<std::string> get_old_keys(const HashOfRecords & storage);
+    void remove_keys_storage(const std::vector<std::string> & keys);
+    void remove_keys_db(const std::vector<std::string> & keys);
+    StringPair split_key(const std::string & key);
+    std::vector<std::string> update_storage(const HashOfRecords & storage);
+    void update_db(const HashOfRecords & storage, const std::vector<std::string> & keys_to_refresh);
+    void extract_values(const std::string & lag_name, json_t * root, HashOfRecords & storage);
```

Comment:
It might be good to group related methods together here, but since they all sort of depend on each other it's probably OK.